File: reference-mcp-servers/finance/QuantConnect_mcp-server_3a656b0_python/tests/algorithms/LiveOrders.cs
#region imports
    using QuantConnect;
    using QuantConnect.Algorithm;
    using QuantConnect.Data;
    using QuantConnect.Securities.Crypto;
#endregion
public class JumpingFluorescentYellowBaboon : QCAlgorithm
{
    private Crypto _btc;
    public override void Initialize()
    {
        _btc = AddCrypto("BTCUSD", Resolution.Second);
    }

    public override void OnData(Slice data)
    {
        if (!Portfolio.Invested)
        {
            SetHoldings(_btc.Symbol, 1);
        }
    }
}

File: reference-mcp-servers/finance/QuantConnect_mcp-server_3a656b0_python/tests/algorithms/live_orders.py
# region imports
from AlgorithmImports import *
# endregion


class LiveOrdersTestAlgorithm(QCAlgorithm):

    def initialize(self):
        self._trades = 0
        self._btc = self.add_crypto('BTCUSD', Resolution.SECOND)

    def on_data(self, data: Slice):
        if self._trades >= 10:
            return
        self._trades += 1
        self.set_holdings(self._btc.symbol, 0 if self.portfolio.invested else 1)
